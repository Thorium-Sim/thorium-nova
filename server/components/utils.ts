export type ComponentOmit<T> = Omit<T, "id" | "defaults" | "getDefaults">;

export abstract class Component {
  static id: string;
  static defaults: any;
  static getDefaults?: Function;
}
