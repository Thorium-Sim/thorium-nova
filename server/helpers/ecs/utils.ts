export function fastSplice(
  array: unknown[],
  startIndex: number,
  removeCount: number
) {
  let len = array.length;
  let removeLen = 0;

  if (startIndex >= len || removeCount === 0) {
    return;
  }

  removeCount = startIndex + removeCount > len ? len - startIndex : removeCount;
  removeLen = len - removeCount;

  for (let i = startIndex; i < len; i += 1) {
    array[i] = array[i + removeCount];
  }

  array.length = removeLen;
}
