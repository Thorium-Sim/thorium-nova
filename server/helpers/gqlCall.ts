import {graphql, GraphQLSchema} from "graphql";
import {Maybe} from "type-graphql";

import createSchema from "./buildSchema";
import {GraphQLContext} from "./graphqlContext";

interface Options {
  query: string;
  variables?: Maybe<{
    [key: string]: any;
  }>;
  context?: GraphQLContext;
}

let schema: GraphQLSchema;

export const gqlCall = async ({
  query,
  variables,
  context = {clientId: "test-client"},
}: Options) => {
  if (!schema) {
    schema = await createSchema();
  }
  return graphql({
    schema,
    source: query,
    contextValue: context,
    variableValues: variables,
  });
};
