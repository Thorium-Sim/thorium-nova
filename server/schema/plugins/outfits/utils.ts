import App from "server/app";
import Components from "server/components";
import Entity from "server/helpers/ecs/entity";
import {pubsub} from "server/helpers/pubsub";
import BasePlugin, {getPlugin, publish} from "../basePlugin";

export function getOutfit(options: {
  pluginId?: string;
  outfitId?: string;
  shipId?: string;
  outfitType?: keyof Entity;
}) {
  const {pluginId, outfitId, shipId, outfitType} = options;
  let outfit: Entity | undefined;
  let ship: Entity | undefined;
  let plugin: BasePlugin | undefined;
  if (pluginId && outfitId) {
    plugin = getPlugin(pluginId);
    outfit = plugin.outfits.find(o => o.id === outfitId);
  } else if (pluginId && !outfitId) {
    throw new Error("Cannot query for outfit. 'outfitId' is required.");
  } else if (outfitId && !pluginId) {
    throw new Error("Cannot query for outfit. 'pluginId' is required.");
  } else if (shipId && outfitType) {
    ship = App.activeFlight?.ships.find(s => s.id === shipId);
    outfit = App.activeFlight?.ecs.entities.find(
      e => e[outfitType] && e.shipAssignment?.shipId === shipId
    );
  } else {
    throw new Error(
      "Cannot query for outfit. Both 'shipId' and 'outfitType' or both 'outfitId' and 'pluginId' are required."
    );
  }
  if (!outfit) throw new Error("Cannot find outfit.");
  return {plugin, ship, outfit};
}

export function outfitPublish({
  plugin,
  ship,
  outfit,
}: {
  plugin?: BasePlugin;
  ship?: Entity;
  outfit: Entity;
}) {
  if (plugin) {
    publish(plugin);
    pubsub.publish("pluginOutfits", {
      pluginId: plugin.id,
      outfits: plugin.outfits,
    });
    pubsub.publish("pluginOutfit", {
      pluginId: plugin.id,
      id: outfit.id,
      outfit: outfit,
    });
  }
  pubsub.publish(`${outfit.isOutfit?.outfitType}Outfit`, {
    outfit,
    pluginId: plugin?.id,
    outfitId: outfit.id,
    shipId: ship?.id,
  });
}

export function updateOutfit({
  pluginId,
  outfitId,
  shipId,
  outfitType,
  update,
}: {
  pluginId?: string;
  outfitId?: string;
  shipId?: string;
  outfitType: keyof Components;
  update: Partial<Components[typeof outfitType]>;
}) {
  const {outfit, ship, plugin} = getOutfit({
    pluginId,
    outfitId,
    shipId,
    outfitType,
  });
  outfit.updateComponent(outfitType, update);
  outfitPublish({plugin, ship, outfit});
  return outfit;
}

export function getAnyOutfit(outfitId: string) {
  return App.plugins.reduce((acc: Entity[], p) => {
    const outfit = p.outfits.find(o => o.id === outfitId);
    if (outfit) {
      acc.push(outfit);
    }
    return acc;
  }, [])[0];
}
