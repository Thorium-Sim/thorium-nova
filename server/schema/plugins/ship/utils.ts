import App from "server/app";
import Components from "server/components";
import Entity from "server/helpers/ecs/entity";
import {pubsub} from "server/helpers/pubsub";
import BasePlugin, {getPlugin, publish} from "../basePlugin";

export function getShip(options: {pluginId?: string; shipId: string}) {
  const {pluginId, shipId} = options;
  let ship: Entity | undefined;
  let plugin: BasePlugin | undefined;
  if (pluginId) {
    plugin = getPlugin(pluginId);
    ship = plugin.ships.find(o => o.id === shipId);
  } else if (shipId) {
    ship = App.activeFlight?.ships.find(s => s.id === shipId);
  } else {
    throw new Error(
      "Cannot query for ship. Either 'shipId' or both 'pluginId' are required."
    );
  }
  if (!ship) throw new Error("Cannot find ship.");
  return {plugin, ship};
}

export function shipPublish({
  plugin,
  ship,
}: {
  plugin?: BasePlugin;
  ship: Entity;
}) {
  if (plugin) {
    publish(plugin);
    pubsub.publish("pluginShips", {
      pluginId: plugin.id,
      ships: plugin.ships,
    });
    pubsub.publish("pluginShip", {
      pluginId: plugin.id,
      shipId: ship.id,
      ship,
    });
  }
  // TODO: Add mid-flight subscription update.
}

export function updateShip({
  pluginId,
  shipId,
  componentType,
  update,
}: {
  pluginId?: string;
  outfitId?: string;
  shipId: string;
  componentType: keyof Components;
  update: Partial<Components[typeof componentType]>;
}) {
  const {ship, plugin} = getShip({
    pluginId,
    shipId,
  });
  ship.updateComponent(componentType, update);
  shipPublish({plugin, ship});
  return ship;
}
