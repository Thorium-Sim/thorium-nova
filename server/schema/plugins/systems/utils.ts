import App from "server/app";
import {pubsub} from "server/helpers/pubsub";

export function publish(plugin: SystemPlugin) {
  pubsub.publish("systemPlugins", {
    id: plugin.id,
    plugins: App.plugins.systems,
  });
  pubsub.publish("systemPlugin", {
    id: plugin.id,
    plugin,
  });
}
export function getSystemPlugin(id: string) {
  const plugin = App.plugins.systems.find(s => s.id === id);
  if (!plugin) {
    throw new Error("Unable to find that system plugin.");
  }
  return plugin;
}
