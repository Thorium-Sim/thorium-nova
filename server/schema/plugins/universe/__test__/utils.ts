import {gqlCall} from "server/helpers/gqlCall";

export async function getPlanetId() {
  const universe = await gqlCall({
    query: `mutation CreateUniverse {
    universeCreate(name:"Test Universe${Math.random()}") {
      id
    }
  }`,
  });
  const id = universe.data?.universeCreate.id;
  const system = await gqlCall({
    query: `mutation CreateSystem($id:ID!) {
    universeTemplateAddSystem(id:$id, position:{x:0,y:0,z:0}) {
      id
    }
  }`,
    variables: {id},
  });

  const systemId = system.data?.universeTemplateAddSystem.id;

  const planet = await gqlCall({
    query: `mutation CreatePlanet($id:ID!,$systemId:ID!, $classification:String!) {
      universeTemplateAddPlanet(id:$id, systemId:$systemId, classification:$classification) {
        id
      }
    }`,
    variables: {
      id,
      systemId,
      classification: "M",
    },
  });

  const planetId = planet.data?.universeTemplateAddPlanet.id;
  return {id, planetId};
}
