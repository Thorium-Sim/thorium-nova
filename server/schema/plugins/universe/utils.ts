import App from "server/app";
import {pubsub} from "server/helpers/pubsub";
import UniverseTemplate from "server/schema/universe";

export function publish(universe: UniverseTemplate) {
  pubsub.publish("templateUniverses", {
    id: universe.id,
    universes: App.plugins.universes,
  });
  pubsub.publish("templateUniverse", {
    id: universe.id,
    universe,
  });
}
export function getUniverse(id: string) {
  const universe = App.plugins.universes.find(u => u.id === id);
  if (!universe) {
    throw new Error("Unable to find that universe.");
  }
  return universe;
}
