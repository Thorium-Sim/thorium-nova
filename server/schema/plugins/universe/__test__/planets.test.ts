import {gqlCall} from "server/helpers/gqlCall";
import {getPlanetId} from "./utils";

const savedMath = global.Math;

describe("universe planets", () => {
  describe("creating planets", () => {
    beforeEach(() => {
      const mockMath = Object.create(global.Math);
      mockMath.random = () => 0.5;
      global.Math = mockMath;
    });
    afterEach(() => {
      global.Math = savedMath;
    });
    it("should create planets of every class", async () => {
      const planetTypes = [
        "A",
        "B",
        "C",
        "D",
        "E",
        "F",
        "G",
        "H",
        "I",
        "J",
        "K",
        "L",
        "M",
        "N",
        "O",
        "P",
      ];
      const universe = await gqlCall({
        query: `mutation CreateUniverse {
        universeCreate(name:"Test Universe") {
          id
        }
      }`,
      });
      const id = universe.data?.universeCreate.id;
      const system = await gqlCall({
        query: `mutation CreateSystem($id:ID!) {
        universeTemplateAddSystem(id:$id, position:{x:0,y:0,z:0}) {
          id
        }
      }`,
        variables: {id},
      });

      const systemId = system.data?.universeTemplateAddSystem.id;

      await gqlCall({
        query: `mutation CreateStar($id:ID!, $systemId:ID!) {
        universeTemplateAddStar(id:$id,systemId:$systemId, spectralType:"G") {
          id
        }
      }`,
        variables: {id, systemId},
      });
      await gqlCall({
        query: `mutation CreateStar($id:ID!, $systemId:ID!) {
        universeTemplateAddStar(id:$id,systemId:$systemId, spectralType:"G") {
          id
        }
      }`,
        variables: {id, systemId},
      });

      for (let type of planetTypes) {
        const planet = await gqlCall({
          query: `mutation CreatePlanet($id:ID!,$systemId:ID!, $classification:String!) {
            universeTemplateAddPlanet(id:$id, systemId:$systemId, classification:$classification) {
            id
            isPlanet {
              classification
              age
              radius
              terranMass
              habitable
              lifeforms
            }
            satellite {
              axialTilt
              distance
              orbitalArc
              orbitalInclination
              eccentricity
              showOrbit
            }
            temperature {
              temperature
            }
          }
          }`,
          variables: {
            id,
            systemId,
            classification: type,
          },
        });
        const {
          id: templateId,
          ...data
        } = planet.data?.universeTemplateAddPlanet;
        expect(data).toMatchSnapshot();
      }
    });
  });
  it("should modify the temperature of the planet", async () => {
    const {id, planetId} = await getPlanetId();
    const planet = await gqlCall({
      query: `mutation ChangePlanet($id:ID!, $objectId:ID!, $temperature:Float!) {
      universeTemplatePlanetSetTemperature(id:$id, objectId:$objectId, temperature:$temperature) {
        id 
        temperature {
          temperature
        }
      }
    }`,
      variables: {id, objectId: planetId, temperature: 0},
    });
    expect(
      planet.data?.universeTemplatePlanetSetTemperature.temperature.temperature
    ).toEqual(0);
  });
  it("should modify the age of the planet", async () => {
    const {id, planetId} = await getPlanetId();
    const planet = await gqlCall({
      query: `mutation ChangePlanet($id:ID!, $objectId:ID!, $age:Float!) {
      universeTemplatePlanetSetAge(id:$id, objectId:$objectId, age:$age) {
        id 
        isPlanet {
          age
        }
      }
    }`,
      variables: {id, objectId: planetId, age: 1337},
    });
    expect(planet.data?.universeTemplatePlanetSetAge.isPlanet.age).toEqual(
      1337
    );
  });
  it("should modify the radius of the planet", async () => {
    const {id, planetId} = await getPlanetId();
    const planet = await gqlCall({
      query: `mutation ChangePlanet($id:ID!, $objectId:ID!, $radius:Float!) {
      universeTemplatePlanetSetRadius(id:$id, objectId:$objectId, radius:$radius) {
        id 
        isPlanet {
          radius
        }
      }
    }`,
      variables: {id, objectId: planetId, radius: 1337},
    });
    expect(
      planet.data?.universeTemplatePlanetSetRadius.isPlanet.radius
    ).toEqual(1337);
  });
  it("should modify the terran mass of the planet", async () => {
    const {id, planetId} = await getPlanetId();
    const planet = await gqlCall({
      query: `mutation ChangePlanet($id:ID!, $objectId:ID!, $terranMass:Float!) {
      universeTemplatePlanetSetTerranMass(id:$id, objectId:$objectId, terranMass:$terranMass) {
        id 
        isPlanet {
          terranMass
        }
      }
    }`,
      variables: {id, objectId: planetId, terranMass: 1337},
    });
    expect(
      planet.data?.universeTemplatePlanetSetTerranMass.isPlanet.terranMass
    ).toEqual(1337);
  });
  it("should modify the lifeforms of the planet", async () => {
    const {id, planetId} = await getPlanetId();
    const planet = await gqlCall({
      query: `mutation ChangePlanet($id:ID!, $objectId:ID!, $lifeforms:String!) {
      universeTemplatePlanetSetLifeforms(id:$id, objectId:$objectId, lifeforms:$lifeforms) {
        id 
        isPlanet {
          lifeforms
        }
      }
    }`,
      variables: {id, objectId: planetId, lifeforms: "1337"},
    });
    expect(
      planet.data?.universeTemplatePlanetSetLifeforms.isPlanet.lifeforms
    ).toEqual("1337");
  });
  it("should modify the habitable status of the planet", async () => {
    const {id, planetId} = await getPlanetId();
    const planet = await gqlCall({
      query: `mutation ChangePlanet($id:ID!, $objectId:ID!, $habitable:Boolean!) {
      universeTemplatePlanetSetHabitable(id:$id, objectId:$objectId, habitable:$habitable) {
        id 
        isPlanet {
          habitable
        }
      }
    }`,
      variables: {id, objectId: planetId, habitable: false},
    });
    expect(
      planet.data?.universeTemplatePlanetSetHabitable.isPlanet.habitable
    ).toEqual(false);
    const planet2 = await gqlCall({
      query: `mutation ChangePlanet($id:ID!, $objectId:ID!, $habitable:Boolean!) {
      universeTemplatePlanetSetHabitable(id:$id, objectId:$objectId, habitable:$habitable) {
        id 
        isPlanet {
          habitable
        }
      }
    }`,
      variables: {id, objectId: planetId, habitable: true},
    });
    expect(
      planet2.data?.universeTemplatePlanetSetHabitable.isPlanet.habitable
    ).toEqual(true);
  });
});
