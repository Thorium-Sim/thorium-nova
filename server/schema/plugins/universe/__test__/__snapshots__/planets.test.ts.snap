// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`universe planets creating planets should create planets of every class 1`] = `
Object {
  "isPlanet": Object {
    "age": 1000500000,
    "classification": "A",
    "habitable": false,
    "lifeforms": "None",
    "radius": 2750,
    "terranMass": 0.65,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 2155893221,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 1583,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 2`] = `
Object {
  "isPlanet": Object {
    "age": 5000500000,
    "classification": "B",
    "habitable": false,
    "lifeforms": "None",
    "radius": 2750,
    "terranMass": 0.65,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 77750170,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 415,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 3`] = `
Object {
  "isPlanet": Object {
    "age": 510000000,
    "classification": "C",
    "habitable": false,
    "lifeforms": "None",
    "radius": 2750,
    "terranMass": 0.55,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 2155893221,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 44,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 4`] = `
Object {
  "isPlanet": Object {
    "age": 510000000,
    "classification": "D",
    "habitable": false,
    "lifeforms": "None",
    "radius": 275,
    "terranMass": 0.21,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 273,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 5`] = `
Object {
  "isPlanet": Object {
    "age": 1000500000,
    "classification": "E",
    "habitable": false,
    "lifeforms": "None",
    "radius": 6250,
    "terranMass": 0.8,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 1122,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 6`] = `
Object {
  "isPlanet": Object {
    "age": 2000000000,
    "classification": "F",
    "habitable": false,
    "lifeforms": "None",
    "radius": 6250,
    "terranMass": 0.8,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 700,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 7`] = `
Object {
  "isPlanet": Object {
    "age": 3500000000,
    "classification": "G",
    "habitable": false,
    "lifeforms": "Primitive single-celled organisms",
    "radius": 6250,
    "terranMass": 1,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 500,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 8`] = `
Object {
  "isPlanet": Object {
    "age": 7000000000,
    "classification": "H",
    "habitable": false,
    "lifeforms": "Primitive single-celled organisms",
    "radius": 5750,
    "terranMass": 0.8,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 77750170,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 700,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 9`] = `
Object {
  "isPlanet": Object {
    "age": 6000000000,
    "classification": "I",
    "habitable": false,
    "lifeforms": "Floating single-celled lifeforms",
    "radius": 2535000,
    "terranMass": 650,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 2155893221,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 234,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 10`] = `
Object {
  "isPlanet": Object {
    "age": 6000000000,
    "classification": "J",
    "habitable": false,
    "lifeforms": "Floating single-celled lifeforms",
    "radius": 47500,
    "terranMass": 55,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 2155893221,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 234,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 11`] = `
Object {
  "isPlanet": Object {
    "age": 7000000000,
    "classification": "K",
    "habitable": true,
    "lifeforms": "Primitive single-celled organisms",
    "radius": 3750,
    "terranMass": 0.8,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 210,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 12`] = `
Object {
  "isPlanet": Object {
    "age": 7000000000,
    "classification": "L",
    "habitable": true,
    "lifeforms": "Hardy plant life",
    "radius": 5000,
    "terranMass": 0.8,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 240,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 13`] = `
Object {
  "isPlanet": Object {
    "age": 6500000000,
    "classification": "M",
    "habitable": true,
    "lifeforms": "Varied and extensive vegetation and animal life, humanoids",
    "radius": 6250,
    "terranMass": 1,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 296,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 14`] = `
Object {
  "isPlanet": Object {
    "age": 6500000000,
    "classification": "N",
    "habitable": true,
    "lifeforms": "Aquatic-based vegetation and animal life, humanoids",
    "radius": 6250,
    "terranMass": 1,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 543,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 15`] = `
Object {
  "isPlanet": Object {
    "age": 6500000000,
    "classification": "O",
    "habitable": true,
    "lifeforms": "Aquatic-based vegetation and animal life, humanoids",
    "radius": 6250,
    "terranMass": 1,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 297,
  },
}
`;

exports[`universe planets creating planets should create planets of every class 16`] = `
Object {
  "isPlanet": Object {
    "age": 6500000000,
    "classification": "P",
    "habitable": true,
    "lifeforms": "Hardy vegetation, animal life, humanoids",
    "radius": 6250,
    "terranMass": 1,
  },
  "satellite": Object {
    "axialTilt": 22.5,
    "distance": 228643390,
    "eccentricity": 0.1,
    "orbitalArc": 180,
    "orbitalInclination": 0,
    "showOrbit": true,
  },
  "temperature": Object {
    "temperature": 263,
  },
}
`;
